### PR TITLE
Trace storage queue

### PR DESCRIPTION
**Pull request by SuyogSoti**
For reviewers, there are two things to look for.

1. I add the `@distributed_trace` or `@distributed_trace_async` for every function that will lead to a network call.
1. I have added the DistributedTracingPolicy to the correct place.
1. I propagate the context with `tracing_context.with_current_context` any time a new thread is created. This should work with both `threading` and `ThreadPoolExecutor`. An example is the following:

`future = executor.submit(tracing_context.with_current_context(uploader.process_chunk), chunk)`

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client_async.py [hunk 1]
```diff
@@ -81,5 +82,6 @@ def _create_pipeline(self, credential, **kwargs):
             config.retry_policy,
             config.logging_policy,
             AsyncStorageResponseHook(**kwargs),
+            DistributedTracingPolicy(),
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client_async.py]: This doesn't need to be an async version of the policy?

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py [hunk 2]
```diff
@@ -159,6 +160,7 @@ def from_connection_string(
             kwargs['secondary_hostname'] = secondary
         return cls(account_url, queue=queue, credential=credential, **kwargs) # type: ignore
 
+    @distributed_trace
     def generate_shared_access_signature(
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py]: No network calls in this function.

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py [hunk 3]
```diff
@@ -138,6 +139,7 @@ def from_connection_string(
             kwargs['secondary_hostname'] = secondary
         return cls(account_url, credential=credential, **kwargs)
 
+    @distributed_trace
     def generate_shared_access_signature(
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py]: No network calls in this function

File: sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client_async.py (hunk 1 above)
Comment (annatisch) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client_async.py]: Additionally - I see you've added the policy to the async pipeline, but haven't decorated the async clients?

Comment (SuyogSoti) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_shared/base_client_async.py]: just decorated them

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py (hunk 2 above)
Comment (SuyogSoti) [reply sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py]: gotten rid of those

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py (hunk 3 above)
Comment (SuyogSoti) [reply sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py]: gotten rid of those

**Pull request merged**